src/components/visualisation/pie.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useRef } from "react";
+import Chart from "chart.js/auto";
+import useWindowSize from "../../hooks/useWindow";
+
```

Comment:
can you move this function to some util function

Reply:
It's a hook so I have kept it in the hooks folder. Should I change?